### PR TITLE
Turmeric and friends

### PR DESCRIPTION
**Pull request by wolfolotl**
## About the PR 
Adds a new botany plant, turmeric, and its associated reagent. Turmeric root can be put in the food processor to get curry powder, which can be used in four new recipes.

image

## Why's this needed?
New plants and recipes are always fun, and turmeric the reagent provides an alternative solution to food poisoning vomiting than chicken soup.

## Changelog
```changelog
(u)Waffleloffle
(*)Adds a new botany plant, turmeric, and its associated reagent. Turmeric root can be put in a food processor to make curry powder, which can be used in some new recipes.
```

### CODE REVIEW

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 1]
```diff
@@ -3289,6 +3289,45 @@ datum
...
+			description = "Has a warm, spicy scent and nausea-soothing properties. But if you get this shit on something, the stain's NEVER coming out."
+			reagent_state = SOLID
+			fluid_r = 224
+			fluid_g = 168
+			fluid_b = 12
+			transparency = 255
+
+			reaction_mob(var/mob/M, var/method = TOUCH, var/volume)
+				. = ..()
+				if(method == TOUCH)
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3304]: I would require a minimum volume for these reactions, something that would mean that smoking this stuff isn't likely to turn the whole shuttle a funny color.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 2]
```diff
@@ -3289,6 +3289,45 @@ datum
...
+				return
+
+			reaction_turf(var/turf/T, var/volume)
+				T.color = rgb(rand(175,255),rand(110,169),92)
+				return
+
+			on_mob_life(var/mob/living/M, var/mult = 1)
+				if(!M) M = holder.my_atom
+				if(prob(50))
+					M.nutrition += 1 * mult
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3320]: I don't think nutrition really does anything meaningful.

Comment (wolfolotl) [reply code/modules/chemistry/Reagents-FoodDrink.dm:3320]: I literally copy-pasted this bit from the chicken soup code so I like, kind of have no idea if it does anything myself

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 3]
```diff
@@ -3289,6 +3289,47 @@ datum
...
+
+			reaction_obj(var/obj/O, var/volume)
+				if(volume >= 10)
+					O.color = rgb(rand(175,255),rand(110,169),92)
+					return
+
+			reaction_turf(var/turf/T, var/volume)
+				if(volume >= 20)
+					T.color = rgb(rand(175,255),rand(110,169),92)
+					return
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3317]: Better to use color matrices for coloring stuff, a la paint.
Example of such in `/datum/statusEffect/dripping_paint`

**Review by Tarmunora — CHANGES_REQUESTED** [review]

**Comment by DisturbHerb** [comment]
These spacepeople must have *great* endurance to eat a whole pot of curry on their own. Stellar work.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 4]
```diff
@@ -3289,6 +3289,59 @@ datum
...
+						0.5, 0.35, 0.0625)
+					return
+
+			reaction_obj(var/obj/O, var/volume)
+				if(volume >= 10)
+					O.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3319]:
```suggestion
					var/oldcol = O.color
					O.color = list(
						0.5, 0, 0,
						0, 0.5, 0,
						0, 0, 0.5,
						0.5, 0.35, 0.0625)
					O.onVarChanged("color", oldcol, O.color)
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 5]
```diff
@@ -3289,6 +3289,59 @@ datum
...
+
+			reaction_mob(var/mob/M, var/method = TOUCH, var/volume)
+				. = ..()
+				if(method == TOUCH && volume >= 10)
+					boutput(M, "<span class='notice'><b>The chemical stains your skin!</b></span>")
+					M.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3310]:
```suggestion
					var/oldcol = M.color
					M.color = list(
						0.5, 0, 0,
						0, 0.5, 0,
						0, 0, 0.5,
						0.5, 0.35, 0.0625)
					M.onVarChanged("color", oldcol, M.color)
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 6]
```diff
@@ -3289,6 +3289,59 @@ datum
...
+					return
+
+			reaction_turf(var/turf/T, var/volume)
+				if(volume >= 20)
+					T.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
```
Comment (Tarmunora) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3328]:
```suggestion
					var/oldcol = T.color
					T.color = list(
						0.5, 0, 0,
						0, 0.5, 0,
						0, 0, 0.5,
						0.5, 0.35, 0.0625)
					T.onVarChanged("color", oldcol, T.color)
```

**Review by Tarmunora — COMMENTED** [review]
nitpick

**Review by Tarmunora — APPROVED** [review]

File: code/modules/food_and_drink/ingredients.dm [hunk 7]
```diff
@@ -783,3 +783,13 @@ obj/item/reagent_containers/food/snacks/ingredient/pepperoni_log
...
+
+/obj/item/reagent_containers/food/snacks/ingredient/currypowder
+	name = "curry powder"
+	desc = "A bag of curry powder. Smells heavenly."
+	icon_state = "currypowder"
+	amount = 1
+	heal_amt = 0
+	food_color = "#e0a80c"
+	initial_volume = 10
+	initial_reagents = "turmpowder"
```
Comment (mordent-goonstation) [inline code/modules/food_and_drink/ingredients.dm:795]: Might as well give the reagent the ID `"currypowder`", while "turm" is a cut-off version of the word it doesn't let me know what this reagent is if I'm not already looking at it in the context of turmeric.

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 8]
```diff
@@ -3289,6 +3289,65 @@ datum
...
+				if(method == TOUCH && volume >= 10)
+					boutput(M, "<span class='notice'><b>The chemical stains your skin!</b></span>")
+					var/oldcol = M.color
+					M.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
+					M.onVarChanged("color", oldcol, M.color)
+					return
```
Comment (mordent-goonstation) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3313]: Nitpick: no need to return if it's the last line in the proc. Same for `reaction_obj`, `reaction_turf`, and `on_mob_life` below.

**Comment by Kyle2143** [comment]
Don't include the actual recipies for dishes in the changelog. The first changelog bullet is all you need. You should add the recipies to an in-game book if you want them to be readily availible to players.

**Comment by Flaborized** [comment]
Bowl sprites look real off to me, the single color outlines on them look particularly odd and jarring when compared to other things in the game, The shading on it is quite banded, it's made up of uniform segments of the same size which looks off. These two things make them look like they're in a different artstyle and they could use some work IMO. Existing soup pot would likely be good for inspiration, though it's metal so there'd be some differences

Other sprites look fine and good to my eye

**Comment by wolfolotl** [comment]
> You should add the recipies to an in-game book if you want them to be readily availible to players.

got it, would it be better to have this book freely available for chefs or to hide it in the nooks and crannies of maint?

**Review by Tarmunora — APPROVED** [review]

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 9]
```diff
@@ -3289,6 +3289,65 @@ datum
...
+			reaction_obj(var/obj/O, var/volume)
+				if(volume >= 10)
+					var/oldcol = O.color
+					O.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
+					O.onVarChanged("color", oldcol, O.color)
+					return
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3324]:
```suggestion
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 10]
```diff
@@ -3289,6 +3289,65 @@ datum
...
+			reaction_turf(var/turf/T, var/volume)
+				if(volume >= 20)
+					var/oldcol = T.color
+					T.color = list(
+						0.5, 0, 0,
+						0, 0.5, 0,
+						0, 0, 0.5,
+						0.5, 0.35, 0.0625)
+					T.onVarChanged("color", oldcol, T.color)
+					return
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3335]:
```suggestion
```

File: code/modules/chemistry/Reagents-FoodDrink.dm [hunk 11]
```diff
@@ -3289,6 +3289,65 @@ datum
...
+						M.cure_disease(virus)
+						boutput(M,"<span class= 'notice'>You feel a little less ill.</span>")
+					if(probmult(10) && istype(virus.master,/datum/ailment/disease/flu))
+						M.cure_disease(virus)
+						boutput(M,"<span class= 'notice'>You feel a little less ill.</span>")
+					if(probmult(10) && istype(virus.master,/datum/ailment/disease/food_poisoning))
+						M.cure_disease(virus)
+						boutput(M,"<span class= 'notice'>You feel a little less sickly.</span>")
+				..()
+				return
```
Comment (ZeWaka) [inline code/modules/chemistry/Reagents-FoodDrink.dm:3349]:
```suggestion
```

File: code/modules/hydroponics/plants_veg.dm [hunk 12]
```diff
@@ -87,3 +87,18 @@ ABSTRACT_TYPE(/datum/plant/veg)
...
+	starthealth = 40
+	growtime = 80
+	harvtime = 140
+	cropsize = 4
+	harvests = 1
+	isgrass = 1
+	endurance = 3
+	genome = 13
+	commuts = list(/datum/plant_gene_strain/metabolism_slow,/datum/plant_gene_strain/immunity_toxin)
+	assoc_reagents = list("currypowder")
```
Comment (ZeWaka) [inline code/modules/hydroponics/plants_veg.dm:104]:
```suggestion
	assoc_reagents = list("currypowder")

```
(someone isn't using the recommended extensions)

**Comment by wolfolotl** [comment]
holy shit why are the dmis breaking it

**Pull request merged**